plugins/content-manager/src/main/java/com/wazuh/contentmanager/index/ContextIndex.java

```diff
@@ -18,303 +18,118 @@
+    public CompletableFuture<SearchResponse> get(String contextName) {
 
-        executor.submit(
-                () -> {
-                    try (ThreadContext.StoredContext ignored =
-                            this.threadPool.getThreadContext().stashContext()) {
-                        searchSourceBuilder.query(null);
-                        SearchRequest searchRequest =
-                                createSearchRequest(this.searchSourceBuilder.trackTotalHits(true));
-
-                        final SearchResponse searchResponse = this.client.search(searchRequest).actionGet();
+        final TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery("_id", contextName);
 
-                        log.info("Result SEARCH ALL: {}", searchResponse.toString());
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(termQueryBuilder);
 
-                        final RestStatus restStatus = searchResponse.status();
+        SearchRequest searchRequest = new SearchRequest(CONTEXTS_INDEX);
+        searchRequest.source(searchSourceBuilder);
 
-                        future.complete(restStatus);
+        CompletableFuture<SearchResponse> future = new CompletableFuture<>();
 
+        this.client.search(
+                searchRequest,
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(SearchResponse searchResponse) {
                         log.info(
-                                "Found {} documents",
-                                Objects.requireNonNull(searchResponse.getHits().getTotalHits()).value);
-                    } catch (Exception e) {
-                        log.error("Error creating SearchRequest due to {}", e.getMessage());
-                        future.completeExceptionally(e);
+                                "Retrieved CTI Catalog Context {} from index with status {}",
+                                contextName,
+                                searchResponse.status());
+                        future.complete(searchResponse);
                     }
-                });
 
-        return future;
-    }
-
-    /**
-     * Create a SearchRequest object from a SearchSourceBuilder object.
-     *
-     * @param searchSourceBuilder builder to create the SearchRequest
-     * @return SearchRequest
-     */
-    private SearchRequest createSearchRequest(SearchSourceBuilder searchSourceBuilder) {
-        SearchRequest searchRequest = new SearchRequest(INDEX_NAME);
-        searchRequest.source(searchSourceBuilder);
-
-        return searchRequest;
-    }
-
-    /**
-     * Update an existing document
-     *
-     * @param id of document
-     * @param document to update the source of the previous document
-     * @return UpdateResponse
-     */
-    public CompletableFuture<RestStatus> update(String id, Document document) {
```

Comment:
We are losing this functionality. Bear in mind that we'll need to update the consumer's info during the periodic updates (last_offset, snapshot) and during the patching (offset, last_offset). 

The method's interface would then be `update(String content, String consumer, ConsumerInfo data)`